lib/routemaster/drain/caching.rb

```diff
@@ -28,6 +29,7 @@ def initialize(options = {})
           use Middleware::RootPostOnly
           use Middleware::Authenticate, options
           use Middleware::Parse
+          use Middleware::ExpireCache,  options
```

Comment:
Seeing as you're fixing this concurrency bug, you might want to pick up + close [LT-447](https://deliveroo.atlassian.net/browse/LT-447).